src/main/java/org/graylog/plugins/threatintel/functions/otx/OTXLookupResult.java

```diff
@@ -19,22 +19,33 @@
 import com.google.common.base.Joiner;
 import com.google.common.collect.ForwardingMap;
 import com.google.common.collect.ImmutableMap;
+import org.graylog2.plugin.lookup.LookupResult;
 
 import java.util.Map;
 
 public class OTXLookupResult extends ForwardingMap<String, Object> {
 
+    public static final String LOOKUP_KEY = "key";
+    public static final String MESSAGE = "message";
+    public static final String HAS_ERROR = "has_error";
+    public static final String OTX_THREAT_INDICATED = "otx_threat_indicated";
     private final ImmutableMap<String, Object> results;
 
-    public static final OTXLookupResult EMPTY = new EmptyOTXLookupResult();
-    public static final OTXLookupResult FALSE = new FalseOTXLookupResult();
+    protected static final OTXLookupResult EMPTY = new EmptyOTXLookupResult();
+    protected static final OTXLookupResult FALSE = new FalseOTXLookupResult();
+
+    public static OTXLookupResult buildFromError(LookupResult lookupResult) {
+        return new FalseOTXLookupResult(
+                (String) lookupResult.multiValue().get(LOOKUP_KEY),
+                (String) lookupResult.multiValue().get(MESSAGE));
```

Comment:
Do we want to handle a potential `NullPointer` here, because a `LookupResult` with `hasError` could also be created without key and message. The `NullPointer` will be caught at some point, so you can decide if you want to handle it explicitly. I just wanted to mention it.

Reply:
So far `LookupResult` with `hasError` are always empty. We now add a case, which is guaranteed to include the expected key/value. 
The point is to communicate the cause of the lookup problem, so I don't want to just return something generic in case there is no message. Might as well fail with an exception, since it would be a bug if it ever happens.